parsers/basic.c

```diff
@@ -140,13 +140,13 @@ static int extract_dim (char const *pos, vString * name, BasicKind kind)
 			pos++;
 
 		if (*pos == '\'')
-			return 0; /* break if we are in a comment */
+			break; /* break if we are in a comment */
```

Comment:
You made a separate commit for these changes.
However, I think this change should be part of the commit " basic: add more advanced extraction of dim-like variables".

If you extend the commit introducing `extract_dim`,  or add a new feature to the commit, making a new commit is greatly appreciated. However, it seems that these changes fix bugs in the base commit, " basic: add more advanced extraction of dim-like variables". In such a case, the commits should be unified.